Microsoft.Identity.Web/HttpContextExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace Microsoft.Identity.Web
+{
+    public static class HttpContextExtensions
+    {
+        /// <summary>
+        /// Keep the validated token in associated with the Http request
+        /// </summary>
+        /// <param name="httpContext">Http context</param>
+        /// <param name="token">Token to preserve after the token is validated so that
+        /// it can be used in the actions</param>
+        public static void StoreTokenUsedToCallWebAPI(this HttpContext httpContext, JwtSecurityToken token)
+        {
+            httpContext.Items.Add("JwtSecurityTokenUsedToCallWebAPI", token);
```

Comment:
@kalyankrishna1 @TiagoBrenck : PR feedback addressed. renamed "token" to "JwtSecurityTokenUsedToCallWebAPI"

Reply:
@jmprieur that name seems a little redundant...what about just "JwtUsedToCallWebAPI"?